src/TwigComponent/tests/Fixtures/templates/components/Alert.html.twig

```diff
@@ -0,0 +1,22 @@
+{% props color = 'blue', size = 'md', class = '' %}
+
+{{ cva('alert', {
+    'base': 'rounded-lg',
+    'variants': {
+        'color': {
+            'blue': 'text-blue-800 bg-blue-50 dark:bg-gray-800 dark:text-blue-400',
+            'red': 'text-red-800 bg-red-50 dark:bg-gray-800 dark:text-red-400',
+            'green': 'text-green-800 bg-green-50 dark:bg-gray-800 dark:text-green-400',
+            'yellow': 'text-yellow-800 bg-yellow-50 dark:bg-gray-800 dark:text-yellow-400',
+        },
+        'size': {
+            'sm': 'px-4 py-3 text-sm',
+            'md': 'px-6 py-4 text-base',
+            'lg': 'px-8 py-5 text-lg',
+        }
+    }
+}) }}
+
+<div class="{{ twm('alert', {color, size}, class) }}">
```

Comment:
Don't you think `twm` (for **T**ail**w**ind **M**erge) and `clm` (for **Cl**asses **M**erge) should only accept classes as parameter, and not the variant name/values? (this is what https://github.com/dcastil/tailwind-merge#tailwind-merge does)

IMO those functions have too much responsability.

```suggestion
{% set styles_alert = cva({
    'base': 'rounded-lg',
    'variants': {
        'color': {
            'blue': 'text-blue-800 bg-blue-50 dark:bg-gray-800 dark:text-blue-400',
            'red': 'text-red-800 bg-red-50 dark:bg-gray-800 dark:text-red-400',
            'green': 'text-green-800 bg-green-50 dark:bg-gray-800 dark:text-green-400',
            'yellow': 'text-yellow-800 bg-yellow-50 dark:bg-gray-800 dark:text-yellow-400',
        },
        'size': {
            'sm': 'px-4 py-3 text-sm',
            'md': 'px-6 py-4 text-base',
            'lg': 'px-8 py-5 text-lg',
        }
    }
}) %}


<div class="{{ twm(cva(styles_alert, { color, size }), class, class2, class3) }}">
```

Here, `cva({})` returns an instance of `CVA`, and if you pass it back to `cva` (`cva(styles_alert, { color, size })`, then it will render your CSS classes.

WDYT?

Reply:
I did that at first but Ryan didn't like it 😅

Reply:
If using strings does it become an option to pass multiple cva names?
Like:
```
<div class="{{ twm('alert other_cva', {color, size, variant_from_other_cva}, class)">
```
Or is this too advanced for now?

Reply:
I thought about the main question here and after some thinking I would say I agree, in this implementation twm() is only usable together with cva, because it always expect a cva as first parameter, which not only combines two things that are independant from each other, it also means anyone who just wants to use some tw classes in a much simpler component can't use this twm to merge tw classes, but has to use cva with base styles for that.

I wonder what Ryan didn't like about separating those two steps 🤔 
After some thinking, what about something like this:

```twig
{% set other_classes = 'm-4 grid grid-rows-[1fr]' %}
{% set alert = cva({
    'base': 'rounded-lg',
    'variants': {
        'color': {
            'blue': 'text-blue-800 bg-blue-50 dark:bg-gray-800 dark:text-blue-400',
            'red': 'text-red-800 bg-red-50 dark:bg-gray-800 dark:text-red-400',
            'green': 'text-green-800 bg-green-50 dark:bg-gray-800 dark:text-green-400',
            'yellow': 'text-yellow-800 bg-yellow-50 dark:bg-gray-800 dark:text-yellow-400',
        },
        'size': {
            'sm': 'px-4 py-3 text-sm',
            'md': 'px-6 py-4 text-base',
            'lg': 'px-8 py-5 text-lg',
        }
    }
}) %}

{% set alert_classes = cvapply(alert, { color, size }) %}
<div class="{{ twm(altert_classes, other_classes, more_tw_classes) }}">
```

It allows for more flexible usage, one could also use multiple cva, get the classes based on params and then throw all of them into twm, which allows some kind of composing, WDYT?